### PR TITLE
WIP: modify sharding readme

### PR DESCRIPTION
**Pull request by enriquefynn**
Renaming terms and other modifications

### CODE REVIEW

File: sharding/README.md [hunk 1]
```diff
@@ -68,23 +68,23 @@ An approach to solving the scalability trilemma is the idea of blockchain shardi
...
+A sharded blockchain system is made possible by having nodes store “signed metadata” in the main chain of latest changes within each shard chain. Through this, we manage to create a layer of abstraction that tells us enough information about the global, synced state of parallel shard chains. These messages are called **collation headers**, which are specific structures that encompass important information about the chainstate of a shard in question. Collations are created by actors known as **proposer nodes** or _collators_ that are randomly tasked into packaging transactions and “selling” them to collator nodes that are then tasked into adding these collations into particular shards through a **proof of stake** system in a designated period of time.
```
Comment (terenc3t) [inline sharding/README.md]: fix **proposer nodes** or _collators_

File: sharding/README.md [hunk 2]
```diff
@@ -93,23 +93,23 @@ In this new protocol, a block is valid when
...
 
-In practice, the first phase of sharding will not be a complete overhaul of the network, but rather an implementation through a smart contract on the main chain known as the **Validator Manager Contract**. Its responsibility is to manage shards and the sampling of proposed validators from a global validator set and will take responsibility for the global reconciliation of all shard states.
+Casper Proof of Stake (Casper [FFG](https://arxiv.org/abs/1710.09437) and [CBC](https://arxiv.org/abs/1710.09437)) makes this quite trivial because there is already a set of global collators that we can select collator nodes from. The source of randomness needs to be common to ensure that this sampling is entirely compulsory and can’t be gamed by the collators in question.
```
Comment (terenc3t) [inline sharding/README.md]: Are we sharing Sharding's collators with Casper's validators?

Comment (enriquefynn) [reply sharding/README.md]: They are the same, is it not the case? At the end we will have to implement Casper.

Comment (rauljordan) [reply sharding/README.md]: We have a simplified PoS in our implementation where 1 collator deposit = 1 vote. We are also operating under the honest majority assumption in our system.

File: sharding/README.md [hunk 3]
```diff
@@ -290,17 +289,17 @@ function addHeader(int _shardId, uint _expectedPeriodNumber, bytes32 _periodStar
 }
 ```
 
-The `addHeader` function is the most importance in the VMC as it is what provides on-chain verification of collation headers immediately and maintains a canonical ordering of processed collation headers by firing a CollationAdded event log.
+The `addHeader` function is the most important function in the SMC as it provides on-chain verification of collation headers, and maintains a canonical ordering of processed collation headers.
 
 Our current [solidity implementation](https://github.com/prysmaticlabs/geth-sharding/blob/master/sharding/contracts/validator_manager.sol) includes all of these functions along with other utilities important for the our Ruby Release sharding scheme.
```
Comment (terenc3t) [inline sharding/README.md]: change link to `https://github.com/prysmaticlabs/geth-sharding/blob/master/sharding/contracts/sharding_manager.sol`

File: sharding/README.md [hunk 4]
```diff
@@ -259,9 +258,9 @@ function getEligibleProposer(int _shardId, int _period) public view returns(addr
...
 ```
 
-The `getEligibleProposer` function uses a block hash as a seed to pseudorandomly select a signer from the validator set. The chance of being selected should be proportional to the validator's deposit. The function should be able to return a value for the current period or any future up to `LOOKAHEAD_PERIODS` periods ahead.
+The `getEligibleProposer` function uses a block hash as a seed to pseudorandomly select a signer from the collator set. The chance of being selected should be proportional to the collator's deposit. The function should be able to return a value for the current period or any future up to `LOOKAHEAD_PERIODS` periods ahead.
 
-#### Withdrawing From the Validator Set
+#### Withdrawing From the collator Set
 
 ```javascript
 function withdraw(int _validatorIndex) public {
```
Comment (terenc3t) [inline sharding/README.md]: ` function withdraw(int _collatorIndex) public {`

File: sharding/README.md [hunk 5]
```diff
@@ -403,33 +402,33 @@ In the sharding consensus mechanism, it is important to consider that we now hav
...
+The primary incentive for collators is to earn the payouts from the proposers offering them collations within their period.
 
 ## Current Status
 
-Currently, Prysmatic Labs is focusing its initial implementation around the logic of the validator and proposer clients. We have built the command line entrypoints as well as the minimum, required functions of the Validator Manager Contract that is deployed to a local Ethereum blockchain instance. Our validator client is able to subscribe for block headers from the running Geth node and determine when we are selected as an eligible proposer in a given period if we have deposited ETH into the contract.
+Currently, Prysmatic Labs is focusing its initial implementation around the logic of the collator and proposer clients. We have built the command line entrypoints as well as the minimum, required functions of the Sharding Manager Contract that is deployed to a local Ethereum blockchain instance. Our collator client is able to subscribe for block headers from the running Geth node and determine when we are selected as an eligible proposer in a given period if we have deposited ETH into the contract.
```
Comment (terenc3t) [inline sharding/README.md]: selected as an eligible ~~proposer~~ collator

File: sharding/README.md [hunk 6]
```diff
@@ -13,16 +13,16 @@ This document serves as a main reference for Prysmatic Labs' sharding implementa
 -   [Go-Ethereum Sharding Alpha Implementation](#go-ethereum-sharding-alpha-implementation)
     -   [System Architecture](#system-architecture)
     -   [System Start and User Entrypoint](#system-start-and-user-entrypoint)
-    -   [The Validator Manager Contract](#the-validator-manager-contract)
+    -   [The Sharding Manager Contract](#the-sharding-manager-contract)
         -   [Necessary Functionality](#necessary-functionality)
-            -   [Depositing ETH and Becoming a Validator](#depositing-eth-and-becoming-a-validator)
+            -   [Depositing ETH and Becoming a Collator](#depositing-eth-and-becoming-a-collator)
             -   [Determining an Eligible Proposer for a Period on a Shard](#determining-an-eligible-proposer-for-a-period-on-a-shard)
```
Comment (terenc3t) [inline sharding/README.md]: Determining an Eligible ~~Proposer~~ Collator

File: sharding/README.md [hunk 7]
```diff
@@ -157,68 +157,67 @@ Here is a full reference spec explaining how our initial system will function:
...
+Back to the collators, the collator client begins to work by its main loop, which involves the following steps:
 
-2.  _**Check Shards for Eligible Proposer:**_ On incoming headers, the client will interact with the VMC to check if the current validator is an eligible validator for an upcoming periods (only a few minutes notice)
+1.  _**Subscribe to incoming block headers:**_ the client will begin by issuing a subscription over JSON-RPC for block headers from the running geth node.
 
-3.  _**If Validator is Selected, Fetch Proposals from Proposal Nodes and Add Collation Headers to VMC:**_ Once a validator is selected, he/she only has a small timeframe to add collation headers through the VMC, so he/she looks for proposals from proposer nodes and accepts those that offer the highest payouts. The validator then countersigns the collation header, receives the full collation with its body after signing, and adds it to the VMC through PoS consensus.
+2.  _**Check shards for eligible proposer:**_ On incoming headers, the client will interact with the SMC to check if the current collator is an eligible collator for upcoming periods (only a few minutes notice)
```
Comment (terenc3t) [inline sharding/README.md]: Check shards for eligible ~~proposer~~ collator

File: sharding/README.md [hunk 8]
```diff
@@ -259,9 +258,9 @@ function getEligibleProposer(int _shardId, int _period) public view returns(addr
 }
 ```
 
-The `getEligibleProposer` function uses a block hash as a seed to pseudorandomly select a signer from the validator set. The chance of being selected should be proportional to the validator's deposit. The function should be able to return a value for the current period or any future up to `LOOKAHEAD_PERIODS` periods ahead.
+The `getEligibleProposer` function uses a block hash as a seed to pseudorandomly select a signer from the collator set. The chance of being selected should be proportional to the collator's deposit. The function should be able to return a value for the current period or any future up to `LOOKAHEAD_PERIODS` periods ahead.
```
Comment (terenc3t) [inline sharding/README.md]: ~~ getEligibleProposer~~ getEligibleCollator

File: sharding/README.md [hunk 9]
```diff
@@ -321,37 +320,37 @@ Explain how CollationAdded logs will later on be used.
...
 ```
 
-The instance connects to a running geth node via JSON-RPC and calls the deposit function on a deployed, Validator Manager Contract to insert the user into a validator set. Then, we subscribe for updates on incoming block headers and call `getEligibleProposer` on receiving each header. Once we are selected, our client fetches and “purchases” proposed, unsigned collations from a proposals pool created by proposer nodes. The validator client accepts a collation that offer the highest payout, countersigns it, and adds it to the VMC all within that period.
+The instance connects to a running geth node via JSON-RPC and calls the deposit function on a deployed, Sharding Manager Contract to insert the user into a collator set. Then, we subscribe for updates on incoming block headers and call `getEligibleProposer` on receiving each header. Once we are selected, our client fetches and “purchases” proposed, unsigned collations from a proposals pool created by proposer nodes. The collator client accepts a collation that offer the highest payout, countersigns it, and adds it to the SMC all within that period.
```
Comment (terenc3t) [inline sharding/README.md]: ~~ getEligibleProposer~~ getEligibleCollator

File: sharding/README.md [hunk 10]
```diff
@@ -516,32 +515,32 @@ Work in progress.
...
+2.  Collator nodes that accept proposals through an open auction system similar to way transactions fees currently work. These nodes then sign these collations and pass them through the SMC for inclusion into a shard through PoS.
 
-## Selecting Eligible Validators Off-Chain
+## Selecting Eligible Collators Off-Chain
 
-In our current implementation for the Ruby Release, we are selecting validators on-chain by calling the `getEligibleProposer` function from the VMC directly. Justin Drake proposes an alternative scheme that could potentially open up new validator selection mechanisms off-chain through a fork-choice rule in [this](https://ethresear.ch/t/fork-choice-rule-for-collation-proposal-mechanisms/922) post on ETHResearch.
+In our current implementation for the Ruby Release, we are selecting collators on-chain by calling the `getEligibleProposer` function from the SMC directly. Justin Drake proposes an alternative scheme that could potentially open up new collator selection mechanisms off-chain through a fork-choice rule in [this](https://ethresear.ch/t/fork-choice-rule-for-collation-proposal-mechanisms/922) post on ETHResearch.
```
Comment (terenc3t) [inline sharding/README.md]: ~~ getEligibleProposer~~ getEligibleCollator

File: sharding/README.md [hunk 11]
```diff
@@ -516,32 +515,32 @@ Work in progress.
...
+In his own words, this scheme “saves gas when calling addHeader and unlocks the possibility for fancier proposer eligibility functions”. A potential way to do so would be through private collator sampling which is elaborated on below:
 
-_“We now look at the problem of private sampling. That is, can we find a proposal mechanism which selects a single validator per period and provides “private lookahead”, i.e. it does not reveal to others which validators will be selected next?
+_“We now look at the problem of private sampling. That is, can we find a proposal mechanism which selects a single validator [collator] per period and provides “private lookahead”, i.e. it does not reveal to others which validators will be selected next?
 There are various possible private sampling strategies (based on MPCs, SNARKs/STARKs, cryptoeconomic signalling, or fancy crypto) but finding a workable scheme is hard. Below we present our best attempt based on one-time ring signatures. The scheme has several nice properties:
 Perfect privacy: private lookahead and private lookbehind (i.e. the scheme never matches eligible proposers with specific validators)
```
Comment (terenc3t) [inline sharding/README.md]: ~~eligible proposers~~ eligible collators

**Comment by terenc3t** [comment]
Looks good to me. Just need to modify the title to not be wip

**Pull request merged**